tests/test_folium.py

```diff
@@ -230,6 +230,29 @@ def test_choropleth_features(self):
         assert '"fillColor":"a_random_color","fillOpacity":0.123454321' in out_str
         assert '"fillOpacity":0.543212345' in out_str
 
+    def test_choropleth_key_on(self):
+        """Test to make sure that Choropleth function doesn't raises
+        a ValueError when the 'key_on' field is set to a column that might
+        have 0 as a value.
+        """
+        with open(os.path.join(rootpath, 'geo_grid.json')) as f:
+            geo_data = json.load(f)
+        data = pd.DataFrame({'idx': {'0': 0, '1': 1, '2': 2, '3': 3, '4': 4, '5': 5},
+                             'value': {'0': 78.0, '1': 39.0, '2': 0.0, '3': 81.0, '4': 42.0, '5': 68.0}})
+        fill_color = 'BuPu'
+        columns = ['idx', 'value']
+        key_on = 'feature.properties.idx'
+
+        try:
```

Comment:
This test is good and indeed fails without the change `Choropleth` class you made.

I do think it's better to remove the try/except. It doesn't add anything. If you want to make it explicit that we expect the call to `Choropleth` NOT to raise an exception, you could do something like [this](https://stackoverflow.com/a/20275035), but IMO it's not necessary.

Reply:
It's done @Conengmo! :) I removed the try/except block since, as you said, it doesn't add anything.